main.sh

```diff
@@ -16,4 +16,106 @@
 #
 # Copyright 2023 The Kepler Contributors
 #
-./common.sh $1
+
+set -eu -o pipefail
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r PROJECT_ROOT
+declare -r BIN_DIR="$PROJECT_ROOT/tmp/bin"
+
+source "$PROJECT_ROOT/.env"
+source "$PROJECT_ROOT/lib/utils.sh"
+
+declare -r CTR_CMD=${CTR_CMD:-docker}
+declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
+declare -r REGISTRY_PORT=${REGISTRY_PORT:-5001}
+declare -r CONFIG_OUT_DIR=${CONFIG_OUT_DIR:-"_output/generated-manifest"}
+declare -r PROMETHEUS_ENABLE=${PROMETHEUS_ENABLE:-true}
+
+declare -r ESTIMATOR_REPO=${ESTIMATOR_REPO:-quay.io/sustainable_computing_io}
+declare -r MODEL_SERVER_REPO=${MODEL_SERVER_REPO:-quay.io/sustainable_computing_io}
+
+init_output_dir() {
+	rm -rf "${CONFIG_OUT_DIR}"
+	mkdir -p "${CONFIG_OUT_DIR}"
+}
+
+cluster_up() {
+	"${CLUSTER_PROVIDER}_up"
+	is_set "${PROMETHEUS_ENABLE}" && {
+		source "$PROJECT_ROOT/lib/prometheus.sh"
+		deploy_prometheus_operator
+	}
+}
+
+cluster_down() {
+	"$CLUSTER_PROVIDER"_down
+}
+
+print_config() {
+	local cluster_config
+	cluster_config=$("${CLUSTER_PROVIDER}"_print_config)
+
+	cat <<-EOF
+		         Configuration
+		━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+		cluster provider   : $CLUSTER_PROVIDER
+		container runtime  : $CTR_CMD
+		config output dir  : $CONFIG_OUT_DIR
+		registry port      : $REGISTRY_PORT
+		Install prometheus : $PROMETHEUS_ENABLE
+
+		Images:
+		  * Estimator    : $ESTIMATOR_REPO
+		  * Model Server : $MODEL_SERVER_REPO
+
+		$cluster_config
+		━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+	EOF
+}
+
+main() {
+	# ensure that all relative file refs are relative to the project root
+	cd "$PROJECT_ROOT"
+	mkdir -p "$BIN_DIR"
```

Comment:
It shouldn't matter if we create or not, since the none of the providers needs to worry about that and can assume this directory exists and thats where any binary must go